theano/gpuarray/dnn.py

```diff
@@ -1557,6 +1561,75 @@ def make_node(self, dy, sm):
         return Apply(self, [dy, sm], [sm.type()])
 
 
+class GpuDnnReduction(DnnBase):
+    check_input = False
+    _f16_ok = True
+    _cop_num_outputs = 2
+
+    __props__ = ('red_op', 'axis', 'arg', 'acc_dtype', 'dtype')
+
+    params_type = ParamsType(red_op=cudnn.cudnnReduceTensorOp_t,
+                             acc_dtype=cudnn.cudnnDataType_t,
+                             c_axis=uint32_t,
+                             handle=handle_type)
+
+    def __init__(self, red_op, axis, acc_dtype, dtype, arg):
```

Comment:
Keep the order the same as props. Also document them. In particular the name arg isn't clear what this does. I would rename it return_indices. The same as the topk PR. It is more clear what it does.